test/e2e/dra/dra.go

```diff
@@ -401,6 +401,73 @@ var _ = framework.SIGDescribe("node")("DRA", feature.DynamicResourceAllocation,
+				return allocatedResourceClaim, err
+			}).WithTimeout(f.Timeouts.PodDelete).ShouldNot(gomega.HaveField("Status.Allocation", (*resourceapi.AllocationResult)(nil)))
+
+			gomega.Eventually(ctx, func(ctx context.Context) error {
+				var err error
+				scheduledPod, err = b.f.ClientSet.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+				if err != nil && scheduledPod.Spec.NodeName != "" {
+					return fmt.Errorf("expected the test pod %s to exist and to be scheduled on a node: %w", pod.Name, err)
+				}
+				return nil
+			}).WithTimeout(f.Timeouts.PodDelete).Should(gomega.BeNil())
+
+			gomega.Expect(allocatedResourceClaim.Status.Allocation).ToNot(gomega.BeNil())
+			gomega.Expect(allocatedResourceClaim.Status.Allocation.Devices.Results).To(gomega.HaveLen(1))
+
+			ginkgo.By("Setting the device status a first time")
+			allocatedResourceClaim.Status.Devices = append(allocatedResourceClaim.Status.Devices,
+				resourceapi.AllocatedDeviceStatus{
+					Driver:     allocatedResourceClaim.Status.Allocation.Devices.Results[0].Driver,
+					Pool:       allocatedResourceClaim.Status.Allocation.Devices.Results[0].Pool,
+					Device:     allocatedResourceClaim.Status.Allocation.Devices.Results[0].Device,
+					Conditions: []metav1.Condition{{Type: "a", Status: "b", Message: "c", Reason: "d"}},
+					Data:       runtime.RawExtension{Raw: []byte(`{"foo":"bar"}`)},
+					NetworkData: &resourceapi.NetworkDeviceData{
+						InterfaceName: "inf1",
+						Addresses:     []string{"10.9.8.0/24", "2001:db8::/64"},
+						HWAddress:     "bc:1c:b6:3e:b8:25",
+					},
+				})
+			// Updates the ResourceClaim from the driver on the same node as the pod.
+			updatedResourceClaim, err := driver.Nodes[scheduledPod.Spec.NodeName].ExamplePlugin.UpdateStatus(ctx, allocatedResourceClaim)
+			framework.ExpectNoError(err)
+			gomega.Expect(updatedResourceClaim).ToNot(gomega.BeNil())
+			gomega.Expect(updatedResourceClaim.Status.Devices).To(gomega.Equal(allocatedResourceClaim.Status.Devices))
+
+			ginkgo.By("Updating the device status")
+			updatedResourceClaim.Status.Devices[0] = resourceapi.AllocatedDeviceStatus{
+				Driver:     allocatedResourceClaim.Status.Allocation.Devices.Results[0].Driver,
+				Pool:       allocatedResourceClaim.Status.Allocation.Devices.Results[0].Pool,
+				Device:     allocatedResourceClaim.Status.Allocation.Devices.Results[0].Device,
+				Conditions: []metav1.Condition{{Type: "e", Status: "f", Message: "g", Reason: "h"}},
+				Data:       runtime.RawExtension{Raw: []byte(`{"bar":"foo"}`)},
+				NetworkData: &resourceapi.NetworkDeviceData{
+					InterfaceName: "inf2",
+					Addresses:     []string{"10.9.8.1/24", "2001:db8::1/64"},
+					HWAddress:     "bc:1c:b6:3e:b8:26",
+				},
+			}
+			updatedResourceClaim2, err := driver.Nodes[scheduledPod.Spec.NodeName].ExamplePlugin.UpdateStatus(ctx, updatedResourceClaim)
```

Comment:
can you do also a Get and compare the output so we exercise that endpoints and code path too?

Reply:
Also instead of doing to UpdateStatus using ServiceSideApply or a Patch for one of them will be great

Reply:
I added the Get, but otherwise, wouldn't it be better as an integration test?

Reply:
yeah, this is because the e2e tests  provide API coverage checking the apiserver endpoints that are exercised in e2e tests, see per example https://github.com/kubernetes/kubernetes/pull/126827

This will exercised all these endpoints in just one test, so we'll not need to add an artificial one like the one. I linked .... it is a nice to have, if you want to follow up or postpone then is ok, I don't want to put too much on this PR, let's keep it this way